velox/functions/prestosql/DateTimeFunctions.h

```diff
@@ -1399,4 +1405,42 @@ struct TimeZoneMinuteFunction : public TimestampWithTimezoneSupport<T> {
   }
 };
 
+template <typename T>
+struct ToISO8601FromTimestampFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  std::string sessionTzName = "";
+  std::string tzOffsetStr = "";
+  const date::time_zone* sessionTimezone = nullptr;
+  int64_t targetTimezoneID = -1;
+
+  FOLLY_ALWAYS_INLINE void initialize(
+      const core::QueryConfig& config,
+      const arg_type<Timestamp>* timestamp) {
+    sessionTimezone = getTimeZoneFromConfig(config);
+    sessionTzName = (sessionTimezone != NULL ? sessionTimezone->name() : NULL);
+  }
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<Varchar>& result,
+      const arg_type<Timestamp>& timestamp) {
+    Timestamp ts = timestamp;
+    TimestampToStringOptions options;
+    options.mode = TimestampToStringOptions::Mode::kFull;
+    options.zeroPaddingYear = true;
+    options.precision = TimestampToStringOptions::Precision::kMilliseconds;
+    auto tsStr = ts.toString(options);
+
+    if (sessionTzName == "UTC") {
+      tzOffsetStr = " UTC";
+    } else if (sessionTimezone != NULL) {
+      auto formatStr = "%Ez";
```

Comment:
This is used in other PRs and so we should make this format specifier a static constant later on.
I suppose there is no helper function to avoid the format specifier?

Reply:
Yes, this same format specifier is used to derive the timezone offset for the `TimestampWithTimezone` signature as well. No helper function - do you think it would be worth making one?

Reply:
Yes. I see you are using the format specifier and if it is used more than once then it would be good to have a helper function that generates the string.

Reply:
Makes sense, I think I will make this change separately as another currently open PR would also be affected - will be easier to deliver once both of these land.